tables/DataMan/test/tDataManInfo.cc

```diff
@@ -0,0 +1,228 @@
+//# tDataManInfo.cc: Test program for class DataManInfo
+//# Copyright (C) 2021
+//# Associated Universities, Inc. Washington DC, USA.
+//#
+//# This program is free software; you can redistribute it and/or modify it
+//# under the terms of the GNU General Public License as published by the Free
+//# Software Foundation; either version 2 of the License, or (at your option)
+//# any later version.
+//#
+//# This program is distributed in the hope that it will be useful, but WITHOUT
+//# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+//# more details.
+//#
+//# You should have received a copy of the GNU General Public License along
+//# with this program; if not, write to the Free Software Foundation, Inc.,
+//# 675 Massachusetts Ave, Cambridge, MA 02139, USA.
+//#
+//# Correspondence concerning AIPS++ should be addressed as follows:
+//#        Internet email: aips2-request@nrao.edu.
+//#        Postal address: AIPS++ Project Office
+//#                        National Radio Astronomy Observatory
+//#                        520 Edgemont Road
+//#                        Charlottesville, VA 22903-2475 USA
+//#
+//# $Id$
+
+#include <casacore/tables/DataMan/DataManInfo.h>
+#include <casacore/tables/Tables/TableDesc.h>
+#include <casacore/tables/Tables/SetupNewTab.h>
+#include <casacore/tables/Tables/ScaColDesc.h>
+#include <casacore/tables/Tables/ArrColDesc.h>
+#include <casacore/casa/Containers/Record.h>
+#include <casacore/casa/Utilities/Assert.h>
+#include <casacore/casa/iostream.h>
+
+#include <casacore/casa/namespace.h>
+using namespace std;
+
+// <summary>
+// Test program for class DataManInfo
+// </summary>
+
+// Test modifying the dminfo record.
+void testDM()
+{
+    TableDesc td;
+    td.addColumn(ScalarColumnDesc<Int>("col1"));
+    td.addColumn(ScalarColumnDesc<Int>("col2"));
+    td.addColumn(ScalarColumnDesc<Int>("col3"));
+    td.addColumn(ArrayColumnDesc<Int>("col4"));
+    // Now create a new table from the description.
+    SetupNewTable aNewTab("tTableCopy_tmp.dm", td, Table::New);
+    Table tabl(aNewTab);
+    Record dminfo = tabl.dataManagerInfo();
+    cout << dminfo;
+    Vector<String> remCols1 =
+      DataManInfo::removeDminfoColumns (dminfo, Vector<String>(1, "col1"), "Standard");
+    cout << dminfo << remCols1 << endl;
+    Vector<String> remCols2 =
+      DataManInfo::removeDminfoColumns (dminfo, Vector<String>(1, "col1"));
+    cout << dminfo << remCols2 << endl;
+    DataManInfo::setTiledStMan (dminfo, Vector<String>(1, "col3"),
+                                "TiledShapeStMan", "TSMData", IPosition(3,3,4,5));
+    cout << dminfo << endl;
+}
+
+void mergeTest()
```

Comment:
The two scopes suggest this test is actually two independent tests: `mergeTestZeroFields()` and `mergeTestOneField`. The same argument holds for the finalizeTest, below.

Reply:
Split them